src/nupic/encoders/md5.cpp

```diff
@@ -0,0 +1,362 @@
+/* MD5
+ converted to C++ class by Frank Thilo (thilo@unix-ag.org)
+ for bzflag (http://www.bzflag.org)
+ 
+   based on:
+ 
+   md5.h and md5.c
+   reference implemantion of RFC 1321
```

Comment:
You would think that md5 and perhaps some other hash modules are not already in the runtime library someplace.  but maybe not since these have copyrights.

Reply:
there's no md5 in std, most use openssl, but I don't want to depend on that moloch (even if it's very common). 
Suggestion: std::hash